test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go

```diff
@@ -136,8 +137,9 @@ func (r *DockerMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}()
 
 	// Handle deleted machines
-	if !dockerMachinePool.ObjectMeta.DeletionTimestamp.IsZero() {
-		return ctrl.Result{}, r.reconcileDelete(ctx, cluster, machinePool, dockerMachinePool)
+	if !dockerMachinePool.DeletionTimestamp.IsZero() {
+		// perform dockerMachinePool delete and requeue in 30s giving time for containers to be deleted.
+		return ctrl.Result{RequeueAfter: 30 * time.Second}, r.reconcileDelete(ctx, cluster, machinePool, dockerMachinePool)
```

Comment:
Thanks for details. Considering we usually resync every 10m, 20m is something unusual.

If the 20m are due to race between creating machines and deleting them, and with this PR we are going to prevent this race to happen, I would prefer to avoid the requeue given that we have a proper fix in place.

If instead the 20m are due to some unknown reason, then we should probably try to root cause (requeing will probably make the test pass, but not fix the underlying issue)

Reply:
There is no race condition, this how the code logic work check [here](https://github.com/kubernetes-sigs/cluster-api/blob/c800c18b26102a71f5ffaa2228a5f3f66ba8fbf4/test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go#L213) in the reconcileDelete, We list the dockerMachines delete them and return (I believe to give time for containers & machines) then next reconcile if there is no dockerMachine we remove the dockerMachinePool finalizer.  So as you said if we rysync in 10m and in the [docker.yaml->wait-delete-cluster](https://github.com/kubernetes-sigs/cluster-api/blob/c800c18b26102a71f5ffaa2228a5f3f66ba8fbf4/test/e2e/config/docker.yaml#L413) we wait 3m for delete so its normal to fail.  Either we increase the docker.yaml->wait-delete-cluster to 15-20m (long time to fail) OR can we set the global resync to 1m during the e2e test ?